include/cuda/stream_ref

```diff
@@ -0,0 +1,183 @@
+#include <cuda/std/detail/__pragma_push>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+/**
+ * \brief A non-owning wrapper for a `cudaStream_t`.
+ *
+ * `stream_view` is a non-owning "view" type similar to `std::span` or
+ * `std::string_view`. \see https://en.cppreference.com/w/cpp/container/span and
+ * \see https://en.cppreference.com/w/cpp/string/basic_string_view
+ *
+ */
+class stream_ref
+{
+private:
+  ::cudaStream_t __stream{0};
+
+public:
+  using value_type = ::cudaStream_t;
+
+  /**
+   * \brief Constructs a `stream_view` of the "default" CUDA stream.
+   *
+   * For behavior of the default stream,
+   * \see
+   * https://docs.nvidia.com/cuda/cuda-runtime-api/stream-sync-behavior.html
+   *
+   */
+  stream_ref() = default;
+
+  /**
+   * \brief Constructs a `stream_view` from a `cudaStream_t` handle.
+   *
+   * This constructor provides implicit conversion from `cudaStream_t`.
+   *
+   * \note: It is the callers responsibilty to ensure the `stream_view` does not
+   * outlive the stream identified by the `cudaStream_t` handle.
+   *
+   */
+  constexpr stream_ref(value_type __stream_) noexcept
+      : __stream{__stream_}
+  {}
+
+  /// Disallow construction from an `int`, e.g., `0`.
+  stream_ref(int) = delete;
+
+  /// Disallow construction from `nullptr`.
+  stream_ref(std::nullptr_t) = delete;
+
+  /**
+   * \brief Compares two `stream_view`s for equality
+   *
+   * \note Allows comparison with `cudaStream_t` due to implicit conversion to
+   * `stream_view`.
+   *
+   * \param lhs The first `stream_view` to compare
+   * \param rhs The second `stream_view` to compare
+   * \return true if equal, false if unequal
+   */
+  _LIBCUDACXX_NODISCARD_ATTRIBUTE friend bool constexpr operator==(stream_ref __lhs, stream_ref __rhs) noexcept
+  {
+    return __lhs.__stream == __rhs.__stream;
+  }
+
+  /**
+   * \brief Compares two `stream_view`s for inequality
+   *
+   * \note Allows comparison with `cudaStream_t` due to implicit conversion to
+   * `stream_view`.
+   *
+   * \param lhs The first `stream_view` to compare
+   * \param rhs The second `stream_view` to compare
+   * \return true if unequal, false if equal
+   */
+  _LIBCUDACXX_NODISCARD_ATTRIBUTE friend bool constexpr operator!=(stream_ref __lhs, stream_ref __rhs) noexcept
+  {
+    return __lhs.__stream != __rhs.__stream;
+  }
+
+  /// Returns the wrapped `cudaStream_t` handle.
+  _LIBCUDACXX_NODISCARD_ATTRIBUTE constexpr value_type get() const noexcept { return __stream; }
+
+  /**
+   * \brief Synchronizes the wrapped stream.
+   *
+   * \throws cuda::cuda_error if synchronization fails.
+   *
+   */
+  void wait() const
+  {
+    const auto __result = ::cudaStreamQuery(get());
+    switch (__result)
+    {
+      case ::cudaSuccess:
+        return;
+      default:
+        ::cudaGetLastError(); // Clear CUDA error state
+#ifndef _LIBCUDACXX_NO_EXCEPTIONS
+        throw cuda::cuda_error{__result, "Failed to synchronize stream."};
+#else
+        ::std::abort();
```

Comment:
Uh-oh. I'm _sure_ that I've said it before when looking at previous iterations of this, but I'm not sure how I feel about invoking host stdlib abort like this. Need to think on this a bunch.

Reply:
Yeah, I am also super icky about that. Suggestions on what to do here would be appreciated. 

It also makes testing this impossible